cloudinit/sources/DataSourceNoCloud.py

```diff
@@ -365,7 +365,12 @@ def _merge_new_seed(cur, seeded):
 class DataSourceNoCloudNet(DataSourceNoCloud):
     def __init__(self, sys_cfg, distro, paths):
         DataSourceNoCloud.__init__(self, sys_cfg, distro, paths)
-        self.supported_seed_starts = ("http://", "https://")
+        self.supported_seed_starts = (
+            "http://",
+            "https://",
+            "ftp://",
+            "stfp://",
```

Comment:
This is a typo, I assume "sftp" was meant, however it really should be "ftps" as "sftp" typically refers to FTP-over-SSH.

Also regarding "ftp" support - unencrypted FTP isn't really considered "safe" these days AFAIK, e.g. most if not all browsers have removed support for it, should "plain" FTP really be added as a supported protocol?

Reply:
> This is a typo, I assume "sftp" was meant, however it really should be "ftps" as "sftp" typically refers to FTP-over-SSH.

Nice catch, thanks. ftps was intended here.

> e.g. most if not all browsers have removed support for it, should "plain" FTP really be added as a supported protocol?

Hmm, good point. The thought was to allow administrators on local trusted networks to use cloud-init with less hassle, but you make a good point about other tooling removing support for ftp - it does look odd to add in 2024. The idea here (docs to come) is that when using "ftps://", only encrypted ftp would work - with unencrypted only allowed if the user uses "ftp://" for their scheme. But you might be right, that we should just add ftps and not bother with unencrypted ftp.

Reply:
I'm not sure I agree about vanilla ftp. Isn't it still commonly used over internal networks? In browsers, support doesn't really buy much functionality other than not having to use another client, so it doesn't justify the maintenance burden. In cloud-init, it could potentially be a lot more useful, and the maintenance is few lines of fallback code.

However, I do suspect that this could be a feature looking for a user, so it's not a strongly held opinion.